src/unix/tty.c

```diff
@@ -34,6 +34,24 @@ static int orig_termios_fd = -1;
 static struct termios orig_termios;
 static uv_spinlock_t termios_spinlock = UV_SPINLOCK_INITIALIZER;
 
+static int uv__tty_is_slave(const int fd) {
+  int result = 0;
+#if defined(__linux__) || defined(__FreeBSD__)
```

Comment:
OpenBSD does definitly not work.

I can't test DragonflyBSD and others though. Would be great if s/o would check if this method works for other Unices too.